### ISSUE TITLE
Mullvad blocks connection to IPv6 server on the same LAN as client

### ISSUE THREAD

**Issue by jm355** [mullvad/mullvadvpn-app#5956]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

I have an IPv6 only server on my home network. For the sake of this ticket to avoid sharing my ip address, let's say my ISP has provided my router with a prefix of 2001:db8::/64, my server has a unique global unicast address of 2001:db8::1, and my laptop has address 2001:db8::2. 

When I try to ping my server while at home connected to mullvad, I get the following:
```
PING my.domain (2001:db8::1) 56 data bytes
From 2001:db8::2 icmp_seq=1 Destination unreachable: Port unreachable
From 2001:db8::2 icmp_seq=2 Destination unreachable: Port unreachable
```
It works as expected when I am connected to mullvad on a different network.

I think the issue is that the routing table recognizes that they have the same prefix, and attempts to route via wlan0 rather than via wg-mullvad, and since the address is not fe80::/10 or fc00::/7, it gets blocked by mullvad thinking it's preventing a leak. 
```
~ $ ip -6 route
2001:db8::/64 dev wlan0 proto ra metric 1024 expires 86394sec mtu 1500 hoplimit 64 pref medium
fc00:bbbb:bbbb:bb01::1 dev wg-mullvad proto static metric 1024 pref medium
fc00:bbbb:bbbb:bb01:5:a:84:feed dev wg-mullvad proto kernel metric 256 pref medium
fe80::/64 dev wlan0 proto kernel metric 256 pref medium
default via fe80::ca8:d00d:fade:7e57 dev wlan0 proto ra metric 1024 expires 594sec mtu 1500 hoplimit 64 pref medium
```

### Expected Behavior

~~Mullvad should recognize that the address has the same prefix as the wlan0 route, count it as "Local network sharing", and not block it.~~

~~If Local network sharing is disabled, mullvad should route an address with the same prefix as wlan0 through the mullvad vpn connection~~

~~(and of course, same for eth0 or any other network interface)~~

edit: mullvad should route this traffic through the vpn like with any other traffic

### Steps to Reproduce

1. Set up ipv6 only server on network 
2. attempt to connect to server from separate device on the same network with the same ipv6 prefix while connected to mullvad vpn

### Failure Logs

```shell
N/A (I think)
```

### Operating system version

Arch Linux

### Mullvad VPN app version

2023.6

**Comment by jm355** [comment]
Ah good to know, I didn't even think about that. In that case, I think the ideal solution would be routing through the vpn. This is what happens on android, I don't run into this issue on android

**Comment by jm355** [comment]
```
~ $ ip -6 route
2001:db8:abcd:1234::/64 dev eth0 proto ra metric 100 expires 86389sec pref medium
fc00:bbbb:bbbb:bb01::1 dev wg0-mullvad proto static metric 1024 pref medium
fc00:bbbb:bbbb:bb01:d::81db dev wg0-mullvad proto kernel metric 256 pref medium
fe80::/64 dev eth0 proto kernel metric 256 pref medium
default via fe80::xxxx:xxxx:xxxx:xxxx dev eth0 proto ra metric 100 expires 589sec pref medium
```
This is my route table where I'm unable to access my server while connected to mullvad on the same network that the server is on. In order to be able to access my server, I had to run `sudo ip -6 route add 2001:db8:abcd:1234::/64 dev wg0-mullvad metric 99 protocol static`, which gave me the following routing table:
```
~ $ ip -6 route
2001:db8:abcd:1234::/64 dev wg0-mullvad proto static metric 99 pref medium
2001:db8:abcd:1234::/64 dev eth0 proto ra metric 100 expires 86389sec pref medium
fc00:bbbb:bbbb:bb01::1 dev wg0-mullvad proto static metric 1024 pref medium
fc00:bbbb:bbbb:bb01:d::81db dev wg0-mullvad proto kernel metric 256 pref medium
fe80::/64 dev eth0 proto kernel metric 256 pref medium
default via fe80::xxxx:xxxx:xxxx:xxxx dev eth0 proto ra metric 100 expires 589sec pref medium
```
With that I'm able to connect. I think that mullvad should automatically add that route for the local network when it connects, so that traffic to a server on the local network doesn't get blocked

**Comment by jm355** [comment]
Closing in favor of https://github.com/mullvad/mullvadvpn-app/issues/7195. In this issue (#5956) I asked for a setting to treat IPs with the same prefix as local traffic and bypass mullvad. @faern pointed out the security issues with that approach, which I appreciate you taking the time to do. But, the issue is still there and it seems like it's been forgotten/buried, so I created #7195 to more concisely ask to route IPs with the same prefix through mullvad, the way the android app does.
